lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -532,7 +532,7 @@ public function getAlterTableSQL(TableDiff $diff)
                     );
                 } elseif ($hasFromComment && ! $hasComment) {
                     $commentsSql[] = $this->getDropColumnCommentSQL($diff->name, $column->getQuotedName($this));
-                } elseif ($hasComment) {
+                } elseif ($hasComment && ! $hasFromComment) {
```

Comment:
Please change to `! $hasFromComment && $hasComment` for the order consistency with the previous conditions.